test/package_installation_test.rb

```diff
@@ -0,0 +1,84 @@
+#! /usr/bin/env rspec
+
+require_relative "./test_helper"
+
+Yast.import "PackageInstallation"
+
+describe Yast::PackageInstallation do
+  subject { Yast::PackageInstallation }
+
+  describe "#Commit" do
+    let(:config) { {"medium_nr" => 0} }
+    let(:result) { [1, [], [], [], []] }
+
+    before do
+      allow(Yast::PackageSlideShow).to receive(:SetCurrentCdNo)
+      allow(Yast::Pkg).to receive(:Commit).with(config)
+        .and_return(result)
+      allow(Yast::PackagesUI).to receive(:show_update_messages)
+    end
+
+    context "when running in normal mode" do
+      it "shows a summary" do
+        expect(Yast::PackagesUI).to receive(:SetPackageSummary)
+        subject.Commit(config)
+      end
+
+      it "returns the commit result" do
+        allow(Yast::PackagesUI).to receive(:SetPackageSummary)
+        expect(subject.Commit(config)).to eq(result)
+      end
+    end
+
+    context "when update messages are received" do
+      let(:result) { [1, [], [], [], [message]] }
+      let(:message) do
+        {
+          "solvable"         => "dummy-package",
+          "text"             => "Some dummy text.",
+          "installationPath" => "/var/adm/update-message/dummy-package-1.0",
+          "currentPath"      => "/var/adm/update-message/dummy-package-1.0",
+        }
+      end
+
+      it "shows the update messages" do
+        expect(Yast::PackagesUI).to receive(:show_update_messages).with(result)
+        subject.Commit(config)
+      end
+
+      context "in installation mode" do
```

Comment:
NP: the contexts are a bit mixed IMHO:

- normal mode
- update messages received
  - installation mode
- fails

I'd rather expect this structure:

- normal mode
  - fails
- installation mode
  - update messages received
  - fails

(You can share the tests for the "fails" part.)

Or alternatively

- fails
  - normal mode
  - installation mode
- update messages received
  - normal mode
  - installation mode

I'm not sure which one is better but I'd rather avoid mixing the contexts if possible...